### PR TITLE
[stdlib] Mark UnicodeScalar.utf16 and UnicodeScalar.UTF16View as public

### PR DESCRIPTION
**Pull request by eliperkins**
`UnicodeScalar.utf16` and `UnicodeScalar.UTF16View` do not have access modifiers and are therefore `internal`. This marks them as `public`.

Resolves [SR-2627](https://bugs.swift.org/browse/SR-2627).

### CODE REVIEW

**Comment by eliperkins** [comment]
@swift-ci Please smoke test

**Comment by modocache** [comment]
@swift-ci please smoke test

**Comment by gottesmm** [comment]
@moiseev This is an API change, no? What is the policy around this?

**Comment by moiseev** [comment]
Thanks @gottesmm. This is indeed an API change and as such should go through the evolution. On the other hand, it can be considered a bug-fix.
@eliperkins can you initiate a swift-evolution discussion?

File: stdlib/public/core/UnicodeScalar.swift [hunk 1]
```diff
@@ -370,29 +370,29 @@ extension UnicodeScalar : Comparable {
 }
 
 extension UnicodeScalar {
-  struct UTF16View {
-    var value: UnicodeScalar
+  public struct UTF16View {
+    public var value: UnicodeScalar
```
Comment (moiseev) [inline stdlib/public/core/UnicodeScalar.swift]: This property should remain internal, actually.

**Comment by eliperkins** [comment]
> can you initiate a swift-evolution discussion?

@moiseev Sure thing! Better to kick it off on the mailing list or just by opening up a proposal on swift-evolution?

**Comment by moiseev** [comment]
Mailing list. If it ends up being considered a bug-fix, it might not even need a formal proposal.
Thanks!

File: stdlib/public/core/UnicodeScalar.swift (hunk 1 above)
Comment (eliperkins) [inline stdlib/public/core/UnicodeScalar.swift]: Addressed in 4a6d368

**Comment by eliperkins** [comment]
Kicked off the email on swift-evolution. Thanks!

**Comment by eliperkins** [comment]
@swift-ci please smoke test

**Comment by dabrahams** [comment]
@swift-ci Please smoke test and merge

**Comment by moiseev** [comment]
@swift-ci Please smoke test and merge

**Comment by gribozavr** [comment]
Could you add some tests?

**Pull request merged**